include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,342 @@
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template< eve::product_type Type
+          , eve::simd_allocator Allocator = eve::aligned_allocator<unsigned char, eve::fixed<64>>
+          >
+  struct soa_vector
+  {
+    public:
+
+    using storage_t = detail::soa_storage<Type,Allocator>;
+
+    //!=============================================================================================
+    //! @name Member types
+    //! @{
+    //==============================================================================================
+    //! Type of data stored in the container
+    using value_type = Type;
+
+    //!   @brief pointer and iterator types.
+    //!   pointer*  - are eve::algo::views::zip_iterator over fields (no conversion to type, just flat fields)
+    //!   iterator* - are a pointer, converter to the Type.
+    //!   They all satisfy eve::algo::relaxed_iterator but not std::iterator
+
+    using pointer               = decltype(storage_t{}.data());
+    using const_pointer         = decltype(std::declval<storage_t const>().data());
+    using pointer_aligned       = decltype(storage_t{}.data_aligned());
+    using const_pointer_aligned = decltype(std::declval<storage_t const>().data_aligned());
+
+    using iterator               = decltype(views::convert(pointer{}              , as<value_type>{}));
+    using const_iterator         = decltype(views::convert(const_pointer{}        , as<value_type>{}));
+    using iterator_aligned       = decltype(views::convert(pointer_aligned{}      , as<value_type>{}));
+    using const_iterator_aligned = decltype(views::convert(const_pointer_aligned{}, as<value_type>{}));
+
+    //==============================================================================================
+    //! @}
+    //==============================================================================================
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    explicit soa_vector(Allocator a = Allocator{}) : storage(a), size_{} {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`.
+    explicit  soa_vector(std::size_t n, Allocator a = Allocator{})
+            : soa_vector(n, value_type{}, a)
+    {}
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type v, Allocator a = Allocator{})
+              : storage(a,n), size_{n}
+    {
+      detail::memory_helper::fill(begin(), 0, n, v );
```

Comment:
alignment is important, don't throw it away!

I should also provide a good way to do aligned parallel arrays, so we could do aligned past the end well.